pkgs/by-name/bl/blivet-gui/package.nix

```diff
@@ -0,0 +1,68 @@
+# Notes for using this package outside of NixOS:
+# 1. --pure cannot be used (as pkexec will be used from the path,
+#    and we can't use nixpkgs polkit due to lack of setuid bit)
+# 2. You must prefix the blivet-gui command with "SHELL=/bin/bash"
+#    (otherwise your system polkit will reject the SHEL Lfrom nixpkgs).
+
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  gtk3,
+  util-linux,
+  gobject-introspection,
+  adwaita-icon-theme,
+  hicolor-icon-theme,
+  wrapGAppsHook3,
+  pkexecPath ? "pkexec",
+  testers,
+  blivet-gui,
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "blivet-gui";
+  version = "2.5.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "storaged-project";
+    repo = pname;
+    rev = version;
+    hash = "sha256-fKd2Vj8clZ6Q7bZipfN5umyMW2rBXMUnpAuDE70p67U=";
+  };
+
+  postPatch = ''
+    substituteInPlace blivetgui/gui_utils.py --replace /usr/share $out/share
+    substituteInPlace blivet-gui --replace "pkexec blivet-gui-daemon" "pkexec $out/bin/blivet-gui-daemon"
+    substituteInPlace blivet-gui --replace "pkexec" "${pkexecPath}"
+  '';
+
+  preFixup = ''
+    makeWrapperArgs+=(--prefix XDG_DATA_DIRS : ${adwaita-icon-theme}/share)
+  '';
```

Comment:
```suggestion
```

Reply:
Unfortunately this still seems to be required, at least on non-NixOS. Without it, startup fails with:

```
(.blivet-gui-wrapped:35127): Gtk-WARNING **: 06:51:22.911: Could not find the icon 'edit-select-all-symbolic'. The 'hicolor' theme
was not found either, perhaps you need to install it.
You can get a copy from:
	http://icon-theme.freedesktop.org/releases
Traceback (most recent call last):
  File "/nix/store/nzg1qlydl5b4392i4n5cvppf0dpw18fk-blivet-gui-2.5.0/bin/..blivet-gui-wrapped-wrapped", line 121, in <module>
    main()
  File "/nix/store/nzg1qlydl5b4392i4n5cvppf0dpw18fk-blivet-gui-2.5.0/bin/..blivet-gui-wrapped-wrapped", line 115, in main
    BlivetGUI(client, exclusive_disks=options.disks, keep_above=options.keep_above, auto_dev_updates=options.auto_dev_updates)
  File "/nix/store/nzg1qlydl5b4392i4n5cvppf0dpw18fk-blivet-gui-2.5.0/lib/python3.12/site-packages/blivetgui/blivetgui.py", line 151, in __init__
    self.list_actions = ListActions(self)
                        ^^^^^^^^^^^^^^^^^
  File "/nix/store/nzg1qlydl5b4392i4n5cvppf0dpw18fk-blivet-gui-2.5.0/lib/python3.12/site-packages/blivetgui/list_actions.py", line 53, in __init__
    icon_edit = Gtk.IconTheme.load_icon(icon_theme, "edit-select-all-symbolic", 16, 0)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: gtk-icon-theme-error-quark: Icon 'edit-select-all-symbolic' not present in theme Adwaita (0)

(.blivet-gui-wrapped:35127): Gtk-CRITICAL **: 06:51:30.069: gtk_main_quit: assertion 'main_loops != NULL' failed
```